### PR TITLE
Fix #12

### LINKED ISSUE #12: Migrate Travis to use docker to test on recent linux distributions
Since the current GNU/Linux distribution supported by Travis is outdated (Ubuntu Trusty), we should update the Travis script to test the library on Ubuntu Xenial or Bionic via docker.

Related to https://github.com/robotology/robotology-superbuild/issues/96

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This Fixes  #12 and `MPCExample.cpp`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,126 +1,104 @@
...
+      -v $HOME/.ccache:$HOME/.ccache \
+      -w $TRAVIS_BUILD_DIR \
+      --env CC \
+      --env CXX \
+      --env TRAVIS_BUILD_DIR \
+      --env TRAVIS_BUILD_TYPE \
+      --env TRAVIS_CMAKE_GENERATOR \
+      --env VALGRIND_TESTS \
+      --env COMPILE_BINDINGS \
+      --env FULL_DEPS \
```
Comment (traversaro) [inline .travis.yml]: I think this three env variables are neither defined in the environment nor used in the scripts running in the Docker container.

File: .ci/script.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+cmake -G"${TRAVIS_CMAKE_GENERATOR}" -DBUILD_GTEST=ON -DBUILD_SHARED_LIBS=ON ..
+cmake --build . --config ${TRAVIS_BUILD_TYPE} --target install
+cd ../..
+
+# osqp
+git clone --recursive https://github.com/oxfordcontrol/osqp.git
+cd osqp
+mkdir build
+cd build
+cmake -G"${TRAVIS_CMAKE_GENERATOR}" -DCMAKE_BUILD_TYPE=${BUILD_TYPE} -DCMAKE_INSTALL_PREFIX=$ROBOT_INSTALL -DUNITTESTS=OFF ..
```
Comment (traversaro) [inline .ci/script.sh]:
```suggestion
cmake -G"${TRAVIS_CMAKE_GENERATOR}" -DCMAKE_BUILD_TYPE=${TRAVIS_BUILD_TYPE} -DCMAKE_INSTALL_PREFIX=$ROBOT_INSTALL -DUNITTESTS=OFF ..
```

File: .ci/install_debian.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+set -e
+
+apt-get update
+
+# noninteractive tzdata ( https://stackoverflow.com/questions/44331836/apt-get-install-tzdata-noninteractive )
+export DEBIAN_FRONTEND=noninteractive
+
+# CI specific packages
+apt-get install -y clang valgrind wget unzip
```
Comment (traversaro) [inline .ci/install_debian.sh]: Yo can probably avoid installing valgrind  if you are not using it for the tests.

**Comment by GiulioRomualdi** [comment]
It seems that Travis is not able to execute `make test` when `Compiler: gcc Xcode: xcode9.4 C++` is chosen https://travis-ci.org/GiulioRomualdi/osqp-eigen/jobs/446635155. This behavior does not happen with Linux (any Ubuntu version) and `Compiler: clang Xcode: xcode9.4 C++` https://travis-ci.org/robotology/osqp-eigen/builds/446635175
@traversaro  do you have any suggestions?

**Comment by traversaro** [comment]
I guess if you have `TRAVIS_CMAKE_GENERATOR="Xcode"` then `make test` is not a valid command, as the generated project is an Xcode project. i suggest to use `ctest` or even better `ctest --output-on-failure --config ${TRAVIS_BUILD_TYPE}` (see for example https://github.com/robotology/idyntree/blob/master/.ci/script.sh#L31).
Note that the `--config` flag passed to the `build` and `test` phases is useful only to support multiple-config generators such as Xcode (https://cgold.readthedocs.io/en/latest/glossary/multi-config.html).

**Comment by GiulioRomualdi** [comment]
I think we can merge this PR

**Pull request merged**